### ISSUE TITLE
Argument of type '{ useNewUrlParser: boolean; useUnifiedTopology: boolean}' is not assignable to parameter of type'

### ISSUE THREAD

**Issue by ssomlk** [Automattic/mongoose#10582]
**_Do you want to request a feature or report a bug?_**

It's a bug

**_What is the current behavior?_**

I am trying to connect my Node.js server to Atlas mongo database. I am using mongoose for this.

```
await mongoose
      .connect(process.env.MONGO_URI!, {
        useNewUrlParser: true,
        useUnifiedTopology: true,
        useCreateIndex: true,
        useFindAndModify: false,
        poolSize: parseInt(process.env.POOL_SIZE!),
      })
      .then((res) => {
        console.log(
          'Connected to Distribution API Database - Initial Connection'
        );
      })
      .catch((err) => {
        console.log(
          `Initial Distribution API Database connection error occured -`,
          err
        );
      });
```

This was working earlier without any issues (I did not update @types/mongoose or mongoose versions at all) 

My dependencies related to this in the package.json file is as below
"dependencies": {
    "@types/mongoose": "5.7.29",
    "mongoose": "5.9.21",
    "typescript": "3.9.5"
  },

And suddenly now I am getting the below error

```
Compilation error in /app/src/index.ts
Using ts-node version 8.10.2, typescript version 3.9.5
[ERROR] 16:25:18 ⨯ Unable to compile TypeScript: src/index.ts(59,11): error TS2769: No overload matches this call.
  Overload 1 of 3, '(uris: string, callback: (err: MongoError) => void): Promise<typeof import("mongoose")>', gave the following error.
    Argument of type '{ useNewUrlParser: boolean; useUnifiedTopology: boolean; useCreateIndex: boolean; useFindAndModify: boolean; poolSize: number; }' is not assignable to parameter of type '(err: MongoError) => void'.
      Object literal may only specify known properties, and 'useNewUrlParser' does not exist in type '(err: MongoError) => void'.
  Overload 2 of 3, '(uris: string, options?: ConnectionOptions | undefined): Promise<typeof import("mongoose")>', gave the following error.
    Argument of type '{ useNewUrlParser: true; useUnifiedTopology: true; useCreateIndex: true; useFindAndModify: false; poolSize: number; }' is not assignable to parameter of type 'ConnectionOptions'.
      Object literal may only specify known properties, and 'poolSize' does not exist in type 'ConnectionOptions'
```

Really appreciate any thoughts on this.

**Comment by ssomlk** [comment]
I started my development one year back and I have not updated the development set up since. But today It started showing up this error even without no updates to neither mongoose or @type/mongoose. I am unable to progress with the project due to this issue

**Comment by hasezoey** [comment]
This is a type issue with `@types/mongoose` (so not an issue to report here), mongoose ships its own definitions since `5.10.19` (yes, a patch version)

the old types (`@types/mongoose`) didnt pin the `@types/mongodb` version, to you probably have version `4.0` of it installed (implicitly), which for that version it is just a stub package (provides no types anymore), because since mongodb `4.0` got rewritten into typescript (now providing its own types)

see https://github.com/Automattic/mongoose/issues/10532#issuecomment-891173320 for more information

TL;DR: upgrade your mongoose version or pin the `@types/mongodb` version

**Comment by vkarpov15** [comment]
@hasezoey we added TS definitions in 5.11.0, not 5.10.19.
